src/components/user/join/Join.jsx

```diff
@@ -0,0 +1,84 @@
+"use client";
+import { useState, useEffect } from "react";
+import Button from "@/components/Button";
+import { useRouter } from "next/navigation";
+import toaster from "@/utils/toaster";
+import { useSession } from "next-auth/react";
+import Fault from "@/utils/error";
+import { api } from "@/utils/api";
+
+const Join = ({ params }) => {
+  const [team, setTeam] = useState(null);
+  const router = useRouter();
+  const { update: sessionUpdate } = useSession();
+
+  const { teamID } = params;
+
+  const handleJoin = () => {
+    api({
+      method: "PUT",
+      url: "/api/members",
+      body: { team: teamID },
+    }).then((response) => {
+      if (response.message !== "OK") {
+        toaster(`${response.message}`, "error");
+        return;
+      }
+      toaster("Successfully joined team!", "success");
+      sessionUpdate({
+        team: teamID,
+      });
+      router.push("/user");
+    });
+  };
+
+  useEffect(() => {
```

Comment:
use effect can be converted into a server component, make the DB call within the component and pass it into a client component for any interaction based things